admin-dev/themes/new-theme/js/components/modal/progress-modal.ts

```diff
@@ -0,0 +1,514 @@
+
+    // Modal header element
+    this.header = document.createElement('div');
+    this.header.classList.add('modal-header');
+    // Modal title element
+    if (params.modalTitle) {
+      this.title = document.createElement('h4');
+      this.title.classList.add('modal-title');
+      this.title.innerHTML = 'Error log';
+      this.header.appendChild(this.title);
+    }
+
+    this.body = document.createElement('div');
+
+    let errorTitle = document.createElement('span');
+    let errorsString = params.modalErrorsOccurred;
+    errorTitle.innerHTML = errorsString.replace('%d', '0');
+    this.body.append(errorTitle);
+    this.errorContainer = document.createElement('div');
+    this.errorContainer.classList.add(componentMap.progressModal.modalErrorContainer);
+    this.errorContainer.classList.add
+    (
+      'd-print-none'
+    );
+    this.body.append(this.errorContainer);
+
+    this.footer = document.createElement('div');
+    this.footer.classList.add('modal-footer');
+    this.switchButton = document.createElement('div');
+    this.switchButton.classList.add(
+      componentMap.progressModal.switchToProgressButton,
+      'btn',
+      'btn-secondary'
+    );
+    this.switchButton.innerHTML = params.modalBackToProcessing;
+
+    this.downloadErrorsButton = document.createElement('div');
+    this.downloadErrorsButton.classList.add(
+      componentMap.progressModal.downloadErrorLogButton,
+      'btn',
+      'btn-secondary'
+    );
+    this.downloadErrorsButton.innerHTML = params.modalDownloadErrorLog;
+
+    this.footer.append(this.switchButton);
+    this.footer.append(this.downloadErrorsButton);
+
+    this.content.append(this.header, this.body, this.footer);
+  }
+}
+
+/**
+ * ConfirmModal component
+ *
+ * @param {InputConfirmModalParams} params
+ * @param {Function} cancelCallback @deprecated You should rely on the closeCallback param
+ */
+export class ProgressModal extends Modal implements ProgressModalType {
+  modal!: ProgressModalParentContainerType;
+  doneCount !: number;
+  total !: number;
+  errors !: Array<string>;
+  cancelCallback !: () => void;
+  progressModal !: ProgressModalContainer;
+  errorModal !: ProgressModalErrorContainer;
+  processStopped !: boolean;
+  params: InputProgressModalParams
+  constructor(
+    inputParams: InputProgressModalParams
+  ) {
+    const params: ProgressModalParams = {
+      id: 'progress-modal',
+      customButtons: [],
+      closable: false,
+      total: 0,
+      dialogStyle: {},
+      ...inputParams,
+    };
+
+    super(params);
+    this.total = params.total;
+    this.errors = [];
+    this.processStopped = false;
+    this.params = params;
+  }
+
+  protected initContainer(params: ProgressModalParams): void {
+    this.modal = new ProgressModalParentContainer(params);
+    super.initContainer(params);
+    this.progressModal = new ProgressModalContainer(params);
+    this.errorModal = new ProgressModalErrorContainer(params);
```

Comment:
```suggestion
```

Reply:
These two view containers are built internally by the `ProgressModalParentContainer` constructor (which should be renamed into `ProgressModalContainer`)